rgw/v2/tests/aws/test_ldap_auth.py

```diff
@@ -0,0 +1,132 @@
+"""
+test_ldap_auth - Test to verify LDAP authentication from RGW
+
+Usage: test_ldap_auth.py
+Polarion ID - CEPH-9793
+
+Operation:
+    Base encode the ldap token from json
+    Query RGW using s3cmd with the base encoded token as the access key
+    Create a bucket on the LDAP user obtained.
+
+"""
+
+
+import argparse
+import base64
+import json
+import logging
+import os
+import random
+import sys
+import traceback
+
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+
+
+from v2.lib import resource_op
+from v2.lib.aws import auth as aws_auth
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.tests.aws import reusable as aws_reusable
+from v2.tests.s3_swift import reusable as s3_reusable
+from v2.utils import utils
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+
+log = logging.getLogger(__name__)
+TEST_DATA_PATH = None
+
+
+def test_exec(config, ssh_con):
+    """
+    Executes test based on configuration passed
+    Args:
+        config(object): Test configuration
+    """
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    # base64 encode json to get ldap token
+    user_data = {
+        "RGW_TOKEN": {
+            "version": 1,
+            "type": "ldap",
+            "id": "ldapuser1",
+            "key": "ldap*user1",
+        }
+    }
+    user_data_str = json.dumps(user_data)
+    token = base64.b64encode(user_data_str.encode("utf-8"))
+    token_str = token.decode("utf-8")
+    log.info(f"LDAP token is {token_str}")
+
+    # Do a s3cmd query with token_str
+    rgw_port = utils.get_radosgw_port_no(ssh_con)
+    rgw_host, rgw_ip = utils.get_hostname_ip(ssh_con)
+    aws_auth.install_aws()
+    cmd = f"AWS_ACCESS_KEY_ID={token_str} AWS_SECRET_ACCESS_KEY=' ' /usr/local/bin/aws s3 ls --endpoint http://{rgw_ip}:{rgw_port}"
+    utils.exec_shell_cmd(cmd)
+
+    cmd = f"radosgw-admin user list"
+    users = utils.exec_shell_cmd(cmd)
+    if "ldapuser1" not in users:
+        raise RGWBaseException("LDAP user not present in RGW user list")
+    # Create a bucket on the LDAP user
+    for bc in range(config.bucket_count):
+        bucket_name = "ldap" + str(bc)
+        cmd = f"AWS_ACCESS_KEY_ID={token_str} AWS_SECRET_ACCESS_KEY=' ' /usr/local/bin/aws s3 mb s3://{bucket_name} --endpoint http://{rgw_ip}:{rgw_port}"
+        out = utils.exec_shell_cmd(cmd)
+        log.info("Bucket created: " + bucket_name)
+
+    # check for any crashes during the execution
+    crash_info = s3_reusable.check_for_crash()
+    if crash_info:
+        raise TestExecError("ceph daemon crash found!")
+
```

Comment:
@TejasC88 , could you also include object creation and listing in this PR

Reply:
@mkasturi18  Object creation and bucket listing added in the script 
Pass log : http://magna002.ceph.redhat.com/ceph-qe-logs/tejas/ldap_auth.txt

Can you pls review ?